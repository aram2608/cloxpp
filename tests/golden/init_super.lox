class A {
  init(x) {
    this.x = x;
  }
}

class B < A {
  init(x, y) {
    super.init(x);
    this.y = y;
  }
}

var b = B(1, 2);
print b.x; // expect 1
print b.y; // expect 2