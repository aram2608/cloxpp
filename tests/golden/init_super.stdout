Token: CLASS class 1
Token: LEFT_BRACE { 1
Token: IDENTIFIER init 2
Token: LEFT_PAREN ( 2
Token: RIGHT_PAREN ) 2
Token: LEFT_BRACE { 2
Token: THIS this 3
Token: DOT . 3
Token: EQUAL = 3
Token: SEMICOLON ; 3
Token: RIGHT_BRACE } 4
Token: RIGHT_BRACE } 5
Token: CLASS class 7
Token: LESS < 7
Token: LEFT_BRACE { 7
Token: IDENTIFIER init 8
Token: LEFT_PAREN ( 8
Token: COMMA , 8
Token: RIGHT_PAREN ) 8
Token: LEFT_BRACE { 8
Token: SUPER super 9
Token: DOT . 9
Token: IDENTIFIER init 9
Token: LEFT_PAREN ( 9
Token: RIGHT_PAREN ) 9
Token: SEMICOLON ; 9
Token: THIS this 10
Token: DOT . 10
Token: EQUAL = 10
Token: SEMICOLON ; 10
Token: RIGHT_BRACE } 11
Token: RIGHT_BRACE } 12
Token: VAR var 14
Token: EQUAL = 14
Token: LEFT_PAREN ( 14
Token: NUMBER 1 14
Token: COMMA , 14
Token: NUMBER 2 14
Token: RIGHT_PAREN ) 14
Token: SEMICOLON ; 14
Token: PRINT print 15
Token: DOT . 15
Token: SEMICOLON ; 15
Token: PRINT print 16
Token: DOT . 16
Token: SEMICOLON ; 16
Token: EOF  16
